pyiceberg/io/__init__.py

```diff
@@ -61,7 +61,7 @@
 AWS_SECRET_ACCESS_KEY = "client.secret-access-key"
 AWS_SESSION_TOKEN = "client.session-token"
 AWS_ROLE_ARN = "aws.role-arn"
-AWS_SESSION_NAME = "aws.session-name"
+AWS_ROLE_SESSION_NAME = "aws.role-session-name"
```

Comment:
This is originally from #1296, which has not been released yet. So we don't need to worry about backwards compatibility 

https://github.com/apache/iceberg-python/pull/1296/files#diff-c08105b6c3a0f63c702051e670b242bb93afdf4146ad1bd19f7205fd7cb39fdfR64